### PR TITLE
create reddit hot threads crawler

### PR DESCRIPTION
**Pull request by Shakthi-Dhar**
### **Describe your change:**
Created a simple python script using praw library to extract data from Reddit

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by Shakthi-Dhar** [comment]
Hi, I have created a python script for users to access/ download the data from Reddit by crawling through the hot threads using the PRAW library.

File: web_programming/reddit_crawler_hot_topics.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+def crawler(subreddit: str) -> list[str]:
+    """
+    >>> crawler("all")
+    ["abcd","hijk","lmn","op","qr"]
+    """
+    data = []
+    hot = reddit.subreddit(subreddit).hot(limit=5)
+    for post in hot:
+        data.append(post.title)
+    return data
```
Comment (cclauss) [inline web_programming/reddit_crawler_hot_topics.py:25]: Please replace with list comprehension as discussed in CONTRIBUTING.md.

File: web_programming/reddit_crawler_hot_topics.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import praw
+
+# creating an instance for praw reddit
+# get all the user cridentials form the reddit page: https://www.reddit.com/prefs/apps
+reddit = praw.Reddit(
+    # Client ID
+    client_id="Enter client id here",
+    # Client Secret Key
```
Comment (cclauss) [inline web_programming/reddit_crawler_hot_topics.py:8]: The code already says this so the comments can be removed.

File: web_programming/reddit_crawler_hot_topics.py (hunk 1 above)
Comment (Shakthi-Dhar) [reply web_programming/reddit_crawler_hot_topics.py:20]: Sure, I shall make the changes as per suggestion

File: web_programming/reddit_crawler_hot_topics.py (hunk 2 above)
Comment (Shakthi-Dhar) [reply web_programming/reddit_crawler_hot_topics.py]: ok

**Comment by Shakthi-Dhar** [comment]
The suggested changes have been made and the code is updated

**Pull request closed without merging**